.evergreen/atlas/setup-variables.sh

```diff
@@ -20,7 +20,15 @@ ATLAS_API_VERSION="v1.0"
 ATLAS_BASE_URL="https://cloud.mongodb.com/api/atlas/$ATLAS_API_VERSION"
 
 # create a unique per CI task tag for the function.
-TASK_NAME=$(echo $task_id | base64 | head -c 8)
+# task_id has the structure $projectID_$variantName_$taskName_$commitHash_$createTime, which is unique per task per ci run per project.
+# task_id is NOT unique across restarts of the same task though, so we include execution to ensure that it is unique for every restart too.
+# 
+# the generated string is then transformed to satisfy Atlas cluster naming requirements
+# - the string cannot be > 64 characters long, so we hash the name to shorten it and maintain its uniqueness (hashes do have collisions but infrequently 
+#   enough that hashing suffices here)
+# - we convert it to hex encoding because cluster names must be alphanumeric (hyphens are allowed)
+transform="process.stdout.write(require('crypto').createHash('md5').update(process.argv[1]).digest().toString('hex'))"
```

Comment:
I couldn't find a hashing CLI tool that would reliably be available across all distros in CI.

Node should be pre-installed on every distro.